deploy/docker/cp-edge/wetty/app.js

```diff
@@ -9,63 +9,130 @@ var child_process = require('child_process');
 const checkMaintenanceMode = require('./maintenance');
 
 // Constants
-var ENV_TAG_RUNID_NAME = 'CP_ENV_TAG_RUNID';
-var CONN_QUOTA_PER_PIPELINE_ID = process.env.CP_EDGE_MAX_SSH_CONNECTIONS || 25;
+const ENV_TAG_RUNID_NAME = 'CP_ENV_TAG_RUNID';
+const CONN_QUOTA_PER_RUN_ID = process.env.CP_EDGE_MAX_SSH_CONNECTIONS || 25;
 
 const CP_MAINTENANCE_SKIP_ROLES = (process.env.CP_MAINTENANCE_SKIP_ROLES || '')
     .split(',')
     .map(o => o.trim().toUpperCase())
     .filter(o => o.length);
 
+function console_log(message) {
+    console.log((new Date()) + ' ' + message);
+}
+
+function console_error(message) {
+    console.error((new Date()) + ' ' + message);
+}
+
 function call_api(api_method, auth_key, httpMethod = 'GET', payload = undefined) {
-    var options = {
+    const options = {
         'headers': {
             'Authorization': 'Bearer ' + auth_key
         },
         json: payload
     };
-    var response = request(httpMethod, process.env.API + api_method, options);
+    const response = request(httpMethod, process.env.API + api_method, options);
     if (response.statusCode == 200) {
         return JSON.parse(response.getBody()).payload;
     }
     return null;
 }
 
-function get_pipe_details(pipeline_id, auth_key) {
-    payload = call_api('/run/' + pipeline_id, auth_key);
-    if (payload) {
-        var parameters = {};
-        if (payload.pipelineRunParameters) {
-            parameters = payload.pipelineRunParameters.reduce(function(parameters, parameter) {
-                parameters[parameter.name] = parameter.value;
-                return parameters;
-            }, {});
+function load_run(run_id, auth_key) {
+    return call_api('/run/' + run_id, auth_key);
+}
+
+function load_runs(pretty_url_path, auth_key) {
+    return call_api(
+        '/run/filter',
+         auth_key,
+        'POST',
+        {
+            'page': 1,
+            'pageSize': 1,
+            'prettyUrl': '%"path":"' + pretty_url_path + '"%'
```

Comment:
Can we change this to something like this:
```
{
  'page': 1,
  'pageSize': 1,
  'prettyUrl': pretty_url_path
}
```
and handle search queries at the backend-side?

Reply:
Thank you for a good advice! I've improved the filter api.